contracts/Sintrop.sol

```diff
@@ -236,6 +251,32 @@ contract Sintrop {
     return canRequest || lastRequestAt == 0;
   }
 
+  function checkExpireData(uint256 inspectionId) internal view returns (bool) {
```

Comment:
Sugestão de código

```solidity
  function checkExpireData(uint256 inspectionId) internal view returns (bool) {
    Inspection memory inspection = inspections[inspectionId];
    uint256 expireBlockAt = inspection.acceptedAt + timeToExpire;

    return block.number > expireBlockAt;
  }
```

Reply:
Legal. Deixa eu ver se entendi aqui:

- A função retorna um bool. No caso a linha 4 com o return, ela vai fazer a verificação de se o bloco atual é maior do que a soma do bloco em que foi aceita a inspeção + a quantidade de blocos para expirar. Se o block.number for maior, returna true. Se a condição não for verdadeira, retorna falso.

Aí nesse caso não deveria ser o contrário? return block.number < expireBlockAt no caso

Reply:
Acredito que é isso mesmo. Alterei aqui e funcionou.

Reply:
Os dois fazem a mesma coisa, acontece que um pergunta se a inspeção já expirou  `block.number > expireBlockAt`
e o outro se a função não expirou  `block.number > expireBlockAt`.

O nome da função poderia ser `expiredInspection` que `block.number > expireBlockAt`. Desse jeito fazemos meio que uma pergunta quando formos utilizar essa função em algum `require`

```solidity
require(!expiredInspection, "Expired inspection")
```